basic_module/distribution/src/core.rs

```diff
@@ -0,0 +1,7 @@
+pub trait Distribution {
+    fn distribute();
+}
+
+pub trait FeeManager {
```

Comment:
It seems like an exported trait for other modules to record fees, doesn't it? Then, the method name should be "record_fee"